food-truck-frontend/src/components/map/route-map/RouteMap.tsx

```diff
@@ -222,17 +226,32 @@ class RouteMapComponent extends React.Component<RouteMapProps, RouteMapState> {
           data: this.mapMultipleFrontendToBackend(this.state.routePts),
           method: "POST",
         })
-        .catch((err) => console.log(err));
+        .catch((err) => {
+          alert(
+            "There was an error saving your changes. Please try again later."
+          );
+          console.log(err);
+          canCont = false;
+        });
 
     // Delete in backend
-    if (this.state.trashedPts.length !== 0)
+    if (canCont && this.state.trashedPts.length !== 0)
```

Comment:
I think this might not work properly. canCont is set to false in the catch block if the last request failed, but my understanding is that the code here gets executed before the catch block executes, because when you do api.request(...), the function execution continues past it while it waits for the response, and when you get the response, it executes the .catch(...) if needed as a callback function. So, the if (canCont) is executed before canCont is set to false.

If my understanding is correct, then you can fix this putting this inside a .then(...) so it only executes if the request was successful.